tests/test_server.py

```diff
@@ -161,6 +161,26 @@ def test_delete_prod_info(self):
         self.assertEqual(status.HTTP_200_OK, response.status_code)
         self.assertEqual(entry_count - 1, self.get_entry_count())
 
+    def test_update_prof_info(self):
+
+        #Test updating existing prod_id
+        test_prod_id = 1
+        prod_info = ProductInformation.find(test_prod_id)
+        if prod_info:
+            data = json.dumps({PROD_NAME :'zen',NEW_QTY :3,RESTOCK_AMT: 7})
+            response = self.app.put(PATH_INVENTORY_PROD_ID.format(test_prod_id),data= data, content_type=JSON)
+            self.assertEqual(status.HTTP_200_OK, response.status_code)
+            new_json = json.loads(response.data)
+            self.assertEqual(new_json['prod_name'], 'zen')
+            self.assertEqual(new_json['new_qty'], 3)
+            self.assertEqual(new_json['restock_amt'],7)
+
+        #Test updating non-exisiting prod_id
+        test_prod_id = 8
+        data = json.dumps({PROD_NAME :'zen',NEW_QTY :3,RESTOCK_AMT : 7})
+        respose = self.app.put(PATH_INVENTORY_PROD_ID.format(test_prod_id),data= data, content_type=JSON)
+        self.assertEqual(status.HTTP_200_OK, response.status_code)
```

Comment:
If test_prod_id doesn't exist in database, shouldn't the response code be 404_BAD_REQUEST?

Reply:
you did the same thing for delete when the prod_id does no exist

Reply:
Delete always return 200OK is for safety issue, we don't want the user to brute force attack us to get all the information in our database. So I guess it's fine to return 200OK for update() also. 
But this is incoherent with line 172 in server.py, where you raise a NotFound error.

Reply:
Is there anything I'm missing? I'm just wondering if a NotFound error is raised, why a 200OK response code passes the test.

Reply:
Yeah it passes the test manually and gives me a NOT FOUND error when prod_id does not exist. 
Let me take a look again and see why this is happening